vortex-array/src/array/chunked/mod.rs

```diff
@@ -234,17 +234,25 @@ impl ValidityVTable<ChunkedArray> for ChunkedEncoding {
     }
 }
 
-impl SubtractScalarFn<ChunkedArray> for ChunkedEncoding {
-    fn subtract_scalar(
+impl BinaryNumericFn<ChunkedArray> for ChunkedEncoding {
+    fn binary_numeric(
         &self,
         array: &ChunkedArray,
-        to_subtract: &Scalar,
-    ) -> VortexResult<ArrayData> {
-        let chunks = array
-            .chunks()
-            .map(|chunk| subtract_scalar(&chunk, to_subtract))
-            .collect::<VortexResult<Vec<_>>>()?;
-        Ok(ChunkedArray::try_new(chunks, array.dtype().clone())?.into_array())
+        rhs: &ArrayData,
+        op: NumericOperator,
+    ) -> VortexResult<Option<ArrayData>> {
+        let mut start = 0;
+
+        let mut new_chunks = Vec::with_capacity(array.nchunks());
+        for chunk in array.chunks() {
+            let end = start + chunk.len();
+            new_chunks.push(binary_numeric(&chunk, &slice(rhs, start, end)?, op)?);
```

Comment:
AFAICT, `new_chunks` will be a Vec<Option<ArrayData>>... we probably want it to be forced to have non-empty chunks, right?

Reply:
Why would it be Vec<Option>?